packages/plugins/src/safe/SafeCompressionPlugin.sol

```diff
@@ -2,42 +2,48 @@
 pragma solidity >=0.7.0 <0.9.0;
 pragma abicoder v2;
 
-import {ECDSA} from "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
 import {HandlerContext} from "safe-contracts/contracts/handler/HandlerContext.sol";
+
 import {IEntryPoint, UserOperation} from "account-abstraction/contracts/interfaces/IEntryPoint.sol";
+import {BLS} from "account-abstraction/contracts/samples/bls/lib/hubble-contracts/contracts/libs/BLS.sol";
+import {IBLSAccount} from "account-abstraction/contracts/samples/bls/IBLSAccount.sol";
 
 import {Safe4337Base, ISafe} from "./utils/Safe4337Base.sol";
 import {WaxLib as W} from "../compression/WaxLib.sol";
 import {IDecompressor} from "../compression/decompressors/IDecompressor.sol";
 
-struct ECDSAOwnerStorage {
-    address owner;
-}
-
-contract SafeCompressionPlugin is Safe4337Base {
-    using ECDSA for bytes32;
+error IncorrectSignatureLength(uint256 length);
 
-    mapping(address => ECDSAOwnerStorage) public ecdsaOwnerStorage;
```

Comment:
Ohh so that's why the mapping was there.

Yeah I think a singleton is a good approach to explore. It adds a bit of complexity overhead but if it solves the double-deployment problem with 4337 initCode then I think that's a good win.